### ISSUE TITLE
Add "Authorized groups" to manage access to products

### ISSUE THREAD

**Issue by pvasilevich** [DefectDojo/django-DefectDojo#2919]
**Is your feature request related to a problem? Please describe**
DefectDojo allows to assign "Authorized users" per product. However, when you have a lot of users (developers) and lot of products, management of such assignments could be pretty tricky.

**Describe the solution you'd like**
It will be great to have concept of "Authorized groups". So user may belongs to any number of groups and product has "Assigned groups" attribute.
If DefectDojo is integrated with external Identity Provider (Azure AD for example), then mapping of logged user to group should be taken from IdP.

**Describe alternatives you've considered**
Provide granular access for every product for every user.

**Additional context**

**Comment by parag85** [comment]
Hello Team,

I am looking for the access option where the user can only update finding status (Reopen,Open,Close)

Is there any option to create a user with above access?

Thanks
Parag Dave

**Comment by reinier-vegter** [comment]
I second this request: we have been looking for a way to grant read/comment permissions to every user (SSO, so that implies every company member).

The only way to do so seems (just getting to know DD in depth, so might be missing something) to be to add everyone to every project as authorized user, e.g. with a nightly API operating script.

It works, but it's not the most elegant workaround, not event driven (after someone's first SSO login, it takes another day before the magic has happened), and the templates don't handle it well ('Contacts' section is showing every user).

I expect it to be a somewhat simple change in the codebase, but I'm not sufficiently familiar with it to do it myself atm.

**Comment by valentijnscholten** [comment]
If you grant superuser permission it will (should) allow read permission on all products.

**Comment by reinier-vegter** [comment]
I'm not sure what the point of IAM and a permissions model is if we just make everyone a superuser.

Couple of scenario's where this is not working:
- I wouldn't want any PO to mark findings as false positive (that needs specialist knowledge, security champions come to mind), but he/she should be able to see metrics, browse findings, make comments etc.
- A business / asset owner: should see metrics and non-tech info, but never change findings.

In other words; not having an informative (read only) role that would suit the above roles, it's harder or not possible at all to use DD as a central security / tracking platform.
Depends a lot on company compliance policy etc, but in my last 3 positions, this wouldn't be an option, because it simply doesn't meet enforced need-to-know principles.
Which in turn limits DD's large potential.

**Comment by valentijnscholten** [comment]
There are lots of issues already about the limited permission model. What I am saying is that making someone superuser is (should be) equal to global readonly.

**Comment by Maffooch** [comment]
#3007 is a step in the right direction. The issue is that Django groups/permissions are not used on the UI whatsoever.

**Comment by StefanFl** [comment]
For a widespread use in my company we will need an enhanced authorisation system as well. Being an IT consultancy we have projects for different customers and internal software. 

- A project team shall be able to see and work with their assigned products, engagements, tests, findings.
- A project team sees only its assigned products / product types --> Need to know principle
- If a project team could create new products for their group, that would be great.

There is a gap currently between staff users (can work with all products) and non-staff users (can see only their products)

**Comment by valentijnscholten** [comment]
New permission system is in the works. If that's ready, we can look at adding groups possibly.

**Comment by StefanFl** [comment]
The foundations were laid with  #4503 and #4530, @Hijerboa is currently working on the UI.

**Comment by valentijnscholten** [comment]
fixed in dev / 2.0.0
